R/sim1sbind.R

```diff
@@ -0,0 +1,58 @@
+#' Simulate nonlinear ligand-receptor binding with one binding site
+#'
+#' A sandbox to simulate and visualize random normal heteroscedastic response data. Variances enlarge with the value of y predicted by the model using a constant coefficeint of variation (cv). The data generating formula is derived from the one-site total binding model: y=Bmax*x/(x+kd). Failure errors in the plot fitting subfunction will occasionally happen due to the random data. These are more frequent with higher cv values. Just re-simulate with modified parameter values.
+#'
+#' @param x a vector of non-exponential linear scale values, usually representing dose or concentration, but can represent any stimulus.
+#' @param bmax the measured value where the receptor population is completely saturated by ligand (i.e. maximum binding)
+#' @param kd the value of x that yields y/Bmax = 0.5 (the equilibrium binding constant)
+#' @param cv the coefficient of variation for y replicates.
+#' @param reps an integer value for number of replicates
+#' @param log logical value. Default is FALSE. If TRUE, linear x values are transformed using a log10 function for plotting. Only for visual aesthetic.
```

Comment:
we might want to omit this option, create a function for log x entry

Reply:
I really don't think we need to do that. I think it's more powerful to keep it wrapped into one function.

The stat_function() call bypasses plotting errors when using geom_smooth(method = "nlsLM") like "singular gradient matrix at start" or "max iterations exceeds 50" or "step halving factor reduced below minimum" (I paraphrased the errors from memory).

Essentially, I use nls to generate estimates for Bmax and Kd (called bmax.var and kd.var). That is what is done in the for loop from lines 37 - 46. Then, stat_function() will plot the model equation and use bmax.var and kd.var and use geom_smooth to curve the line. I've tested it with several different values for x, bmax, kd, and cv, and the stat_function() call modifies the curve accordingly.